### PR TITLE
Bug 1866128 - Add new debug fab icon

### PR DESCRIPTION
**Pull request by MozillaNoah**
image

image

The SVG was optimized before being imported into Android Studio

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1866128

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugOverlay.kt [hunk 1]
```diff
@@ -36,7 +36,7 @@ fun DebugOverlay() {
         modifier = Modifier.fillMaxSize(),
     ) {
         FloatingActionButton(
-            icon = painterResource(R.drawable.ic_new),
+            icon = painterResource(R.drawable.ic_debug_transparent_fire_24),
```
Comment (MatthewTighe) [inline fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugOverlay.kt:39]: I'm not seeing that we include the size of the icon in our other drawables. Do we need it here and can we remove the `_24` if not?

**Review by MatthewTighe — APPROVED** [review]
IDK why but the icon makes it feel so real and official lol. LGTM!

File: fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugOverlay.kt (hunk 1 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugOverlay.kt:39]: so the icons in fenix don't typically have the `_24`, but all of the icons in AC have it. I believe the preference going forward it to include it

Comment (MatthewTighe) [reply fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugOverlay.kt:39]: Gotcha, thanks!

**Pull request merged**